File: capacity-club-api/src/domain-modules/member-card/model/payload/member-card-update.payload.ts
import { UniqueId } from '@common/model/unique-id';
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty } from 'class-validator';
import { Member } from 'domain-modules/member/model';

/**
 * Class used to update member card.
 * This class defines the payload used when updating existing member card.
 */
export class MemberCardUpdatePayload {
  @ApiProperty()
  @IsNotEmpty()
  member_card_id: UniqueId;
  @ApiProperty()
  collective_session_count: number;

  @ApiProperty()
  individual_session_count: number;

  @ApiProperty()
  member: Member;
}

File: capacity_club_mobile_app/lib/domain-features/data/member-card/model/payload/member_card_update_payload.dart
import 'package:capacity_club_mobile_app/core/model/entities/unique_id.dart';
import 'package:equatable/equatable.dart';

class MemberCardUpdatePayload extends Equatable {
  final UniqueId member_card_id;
  final int collective_session_count;
  final int individual_session_count;

  const MemberCardUpdatePayload({
    required this.member_card_id,
    required this.collective_session_count,
    required this.individual_session_count,
  });

  Map<String, dynamic> toJson() => {
        'member_card_id': member_card_id.toJson(),
        'collective_session_count': collective_session_count,
        'individual_session_count': individual_session_count,
      };

  @override
  List<Object?> get props => [
        member_card_id,
        collective_session_count,
        individual_session_count,
      ];
}
